### PR TITLE
[REFACTOR] Refactor the async model workflow with the middle layer

### PR DESCRIPTION
**Pull request by LeaYeh**
1. Changed the EPOLLET to default, means we decide to use level trigger but not edge trigger mode. Because we cannot use fcntl() and with the edge trigger we need to set the fd to non-blocking mode or it would not improve the perf but also slow down the system and cause the I/O blocking. It's super sad nginx also use edge trigger to get the perf.
2. Changed the config to singleton pattern, that's important to break the dependency between the submodules and to make the code more maintainable and readable, anytime and in any class we don't need to pass the config as argument. I will also change the logger to singleton pattern in next PR.
3. Added the RequestProcessor as middle layer to communicate with Analyzer, Builder and ConnectionHandler.
4. Refactored the kernel structure, now the Reacor HAS a ConnectionHandler and the ConnectionHandler handles multiple connections with different clients. That is the major spirit of the reactor pattern, I misunderstood it. and it will also be more make sense for the SRMW structure.
5. Add interface prototype for Analyzer and Builder

### CODE REVIEW

**Review by leske42 — APPROVED** [review]
so huge

**Comment by LeaYeh** [comment]
My bad it's out of my control

File: include/shell/UriAnalyzer.hpp [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
 #include "defines.hpp"
-#include <iostream>
+#include <string>
 
 namespace webshell
 {
 
-class UriAnalyser
+class UriAnalyzer
```
Comment (yenthing) [inline include/shell/UriAnalyzer.hpp:9]: Analyser is not wrong. One is American style, the other is UK

**Review by yenthing — APPROVED** [review]
Boah this one is a lot. And you changed much after last Sunday as you explained me. Could you upload your note with graphs you drew for acceptor, receive and bla blah. The relationship between them

**Comment by LeaYeh** [comment]
image
image
image
image

**Pull request merged**